### ISSUE TITLE
How do I control which AWS profile is used with the automated test project?

### ISSUE THREAD

**Issue by macfarmw** [aws/aws-toolkit-visual-studio#172]
I created an app and used the template: AWS Serverless Application with Tests (.NET Core - C#). I selected the Blog API sample.
This created an xUnit test project in the solution. I added my own lambda function and tests and had everything working. Since then I have changed the IAM account that I want to use from the build agents for CI/CD. Now when the tests run on the CI/CD server they are failing because they are still trying to run under the previous IAM account. I know this because the error message explicitly lists the ARN for the IAM account that does not have the needed permission to perform the operations being called in the test case.

I have confirmed that the named AWS profile is setup correctly on the build agent and I specify this profile in the aws-lambda-tools-defaults.json file. Somehow the tests project is still using the previous account that was active when the project was first created.

Can you please specify how I can control which AWS profile is used by the test project when it runs?

**Comment by awschristou** [comment]
Unless you've modified the test class (FunctionTest), it initializes a DynamoDB client like this:
```
this.DDBClient = new AmazonDynamoDBClient(RegionEndpoint.USWest2);
```

Because credentials are not explicitly provided to the client, the AWS SDK then follows the default [Credential and Profile Resolution](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html#creds-assign). It is likely using the default credentials available to the system that the test is running on.

**Comment by macfarmw** [comment]
That article is very helpful. I tried something that's not mentioned in the article which seems to work fine.

I added this line near the beginning of the test class.
`Environment.SetEnvironmentVariable("AWS_PROFILE", "BuildAgentProfile");`

The BuildAgentProfile named profile is configured on my local PC and on the build agent. With this environment variable set, the remaining commands in the tests work fine now.
